src/lua/protobuf.lua

```diff
@@ -0,0 +1,624 @@
+    integral_only = false,
+    limits = {
+        number = {-3.4028234E+38, 3.4028234E+38},
+    },
+    encode = wireformat.encode_float,
+}
+
+scalars.fixed32 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {0, 2^32 - 1},
+        int64 = {0LL, 4294967295LL},
+        uint64 = {nil, 4294967295ULL},
+    },
+    encode = wireformat.encode_fixed32,
+}
+
+scalars.sfixed32 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {-2^31, 2^31 - 1},
+        int64 = {-2147483648LL, 2147483647LL},
+        uint64 = {nil, 2147483647ULL},
+    },
+    encode = wireformat.encode_fixed32,
+}
+
+scalars.double = {
+    accept_type = 'number',
+    encode_as_packed = true,
+    integral_only = false,
+    limits = {
+        number = {-1.7976931348623157E+308, 1.7976931348623157E+308},
+    },
+    encode = wireformat.encode_double,
+}
+
+scalars.fixed64 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {0, 2^64 - 1},
+        int64 = {0LL, nil},
+    },
+    encode = wireformat.encode_fixed64,
+}
+
+scalars.sfixed64 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {-2^63, 2^63 - 1},
+        int64 = {-9223372036854775807LL, 9223372036854775806LL},
+        uint64 = {nil, 9223372036854775806ULL},
+    },
+    encode = wireformat.encode_fixed64,
+}
+
+scalars.string = {
+    accept_type = 'string',
+    encode_as_packed = false,
+    encode = wireformat.encode_len,
+}
+
+scalars.bytes = scalars.string
+
+scalars.int32 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {-2^31, 2^31 - 1},
+        int64 = {-2147483648LL, 2147483647LL},
+        uint64 = {nil, 2147483647ULL},
+    },
+    encode = wireformat.encode_int,
+}
+
+scalars.sint32 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {-2^31, 2^31 - 1},
+        int64 = {-2147483648LL, 2147483647LL},
+        uint64 = {nil, 2147483647ULL},
+    },
+    encode = wireformat.encode_sint,
+}
+
+scalars.uint32 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {0, 2^32 - 1},
+        int64 = {0LL, 4294967295LL},
+        uint64 = {nil, 4294967295ULL},
+    },
+    encode = wireformat.encode_int,
+}
+
+scalars.int64 = {
+    accept_type = {'number', 'cdata'},
+    encode_as_packed = true,
+    integral_only = true,
+    limits = {
+        number = {-2^63, 2^63 - 1},
+        int64 = {-9223372036854775808LL, 9223372036854775807LL},
+        uint64 = {nil, 9223372036854775807ULL},
+    },
+    encode = wireformat.encode_int,
+}
```

Comment:
`limits.int64` accepts all the `int64_t` values. It seems, a range check for `int64_t` is not needed and we can omit `limits.int64`.

The same for `scalars.sint64.limits.int64`.

Reply:
I agree, removed it.